internal/clients/gcp.go

```diff
@@ -96,17 +96,9 @@ func constructFederatedCredentials(providerID, serviceAccount string) ([]byte, e
 // returns Terraform provider setup configuration
 // NOTE(hasheddan): this function is slightly over our cyclomatic complexity
 // goal. Consider refactoring before adding new branches.
-func TerraformSetupBuilder(version, providerSource, providerVersion string, tfProvider *schema.Provider, scheduler terraform.ProviderScheduler) terraform.SetupFn { //nolint:gocyclo
+func TerraformSetupBuilder(tfProvider *schema.Provider) terraform.SetupFn { //nolint:gocyclo
 	return func(ctx context.Context, client client.Client, mg resource.Managed) (terraform.Setup, error) {
-		ps := terraform.Setup{
```

Comment:
When we leave the TF CLI base reconciliations for every upjet-based provider, do we consider deprecating and removing these setup fields: `Version`, `Requirement`, `Scheduler`?

Reply:
I think we may keep them for the other community upjet-based providers. Maintainers of those providers may prefer to integrate via the CLI because that integration happens at a higher layer.